rust/psibase/src/table.rs

```diff
@@ -124,20 +125,20 @@ impl<Record: TableRecord> Table<Record> {
                 idx.append_key(&mut key_buffer);
 
                 old_key.append_key(&mut key_buffer);
-                kv_remove(self.db_id, &KeyView::new(&key_buffer));
-                key_buffer.truncate(self.prefix.len() + 1);
+                kv_remove(self.db_id(), &KeyView::new(&key_buffer));
+                key_buffer.truncate(self.prefix().len() + 1);
 
                 new_key.append_key(&mut key_buffer);
-                kv_insert_unique_bytes(self.db_id, &KeyView::new(&key_buffer), pk);
-                key_buffer.truncate(self.prefix.len());
+                kv_insert_unique_bytes(self.db_id(), &KeyView::new(&key_buffer), pk);
```

Comment:
Long term, not needed in this PR, it'd be nice to do the dup check before any writes then return an error

Reply:
The C++ code aborts because the error approach we once preferred ended up causing clang to bloat the wasms.